feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java

```diff
@@ -20,165 +34,170 @@
+			String result = EntityUtils.toString(entity);
+			httpClient.close();
+			response.close();
+			return ((result.charAt(0) == '1') ? true : false);
+		}
+	}
+	
+	public  void sendRequestToCreate(CloseableHttpResponse response,List<NameValuePair> nameValuePairs,HttpRequestBase request) throws Exception {
+		
+		URI uri = new URIBuilder(request.getURI()).setParameters(nameValuePairs).build();
+		((HttpRequestBase) request).setURI(uri);
+		CloseableHttpClient httpClient = HttpClients.createDefault();
+		logger.info("URI :{}",uri);
+		request.addHeader("Content-Type", content_type);
+		request.addHeader("Accept", accept);
+		request.addHeader("company", company);
+		request.addHeader("Authorization", "Bearer " + auth);
+		response = httpClient.execute(request);
+		logger.info("Status Code  : {}", response.getStatusLine().getStatusCode());
+		response.close();
+		httpClient.close();
+	}
+	
+	
+	public List<NameValuePair> constructCustomerFromOpenmrsPatient(String name,String uuid,String currencyId,OpenMRSPersonAddress address,List<OpenMRSPersonAttribute> attribute){
+		List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>();
+		nameValuePairs.add(new BasicNameValuePair("name", name));
+		nameValuePairs.add(new BasicNameValuePair("contact_name", uuid));
+		nameValuePairs.add(new BasicNameValuePair("currency_id", currencyId));
+		nameValuePairs.add(new BasicNameValuePair("billing[name]", address.getAddress1()));
+		nameValuePairs.add(new BasicNameValuePair("billing[address_street_1]", address.getAddress2()));
+		nameValuePairs.add(new BasicNameValuePair("billing[address_street_2]", address.getAddress3()));
+		nameValuePairs.add(new BasicNameValuePair("billing[city]", address.getCityVillage()));
+		nameValuePairs.add(new BasicNameValuePair("billing[state]", address.getStateProvince()));
+		nameValuePairs.add(new BasicNameValuePair("billing[country_id]", address.getCountry()));
+		nameValuePairs.add(new BasicNameValuePair("billing[zip]", address.getPostalCode().toString()));
+		for (OpenMRSPersonAttribute openMRSPersonAttribute : attribute) {
+			String display=openMRSPersonAttribute.getAttributeType().getDisplay();
+			String value = openMRSPersonAttribute.getValue().toString();
+			if(display.equals("phoneNumber")) {
+			nameValuePairs.add(new BasicNameValuePair("phone",value));
+			}
+			else
+			nameValuePairs.add(new BasicNameValuePair(display,value));
+		}
+		return nameValuePairs;
+		
+	}
+	
+	public void create_customer(OpenMRSPatientFullRepresentation patientFR, String auth) throws Exception {
+		String name = patientFR.getPerson().getPreferredName().getPreferredFullName();
+		String uuid = getuuid(patientFR);
+		OpenMRSPersonAddress address = patientFR.getPerson().getPreferredAddress();
+		List<OpenMRSPersonAttribute> attribute=patientFR.getPerson().getAttributes();
+		List<NameValuePair>nameValuePairs=constructCustomerFromOpenmrsPatient(name,uuid,currencyId,address,attribute);
+		CloseableHttpResponse response = null;
```

Comment:
Why not also create the RESPONSE object in the sendRequestToCrater( ) method?

Reply:
done